Adyen/Core/Payment Methods/PaymentMethod.swift

```diff
@@ -64,5 +64,7 @@ public func == (lhs: StoredPaymentMethod, rhs: StoredPaymentMethod) -> Bool {
 
 /// :nodoc:
 public func == (lhs: PaymentMethod, rhs: PaymentMethod) -> Bool {
-    lhs.type == rhs.type && lhs.name == rhs.name
+    lhs.type == rhs.type &&
+        lhs.name == rhs.name &&
+        String(describing: type(of: lhs)) == String(describing: type(of: rhs))
```

Comment:
We can use here `type(of: lhs) === type(of: rhs)`

Reply:
no doesn't work